File: Learnify.Service/Implementation/QuizService.cs
using Learnify.Domain.DTO;
using Learnify.Domain.Entities;
using Learnify.Repository.Interfaces;
using Learnify.Service.Interfaces;
using Learnify.Service.Mappers;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Learnify.Service.Implementation
{
    public class QuizService : IQuizService
    {
        private readonly IQuizRepository repository;

        public QuizService(IQuizRepository repository)
        {
            this.repository = repository;
        }

        public IQueryable<Quiz> GetAllQuizzes()
        {
            return repository.GetAll();
        }

        public Quiz GetQuiz(Guid quizId)
        {
            var quiz = repository.Get(quizId);
            return quiz;
        }

        public Quiz GetQuizWithQuestions(Guid quizId)
        {
            var quiz = repository.GetQuizWithQuestions(quizId);
            return quiz;
        }

        public IQueryable<Quiz> GetQuizzesByCourse(Guid courseId)
        {
            var quizzes = repository.GetQuizzesByCourseId(courseId);
            return quizzes;
        }

        public Task InsertQuiz(QuizRequest quizRequest)
        {
            var quiz = QuizMapper.CopyFromDto(quizRequest);
            repository.Insert(quiz);
            return Task.CompletedTask;
        }

        public Task UpdateQuiz(UpdateQuiz updateQuiz)
        {
            var quiz = QuizMapper.CopyFromUpdateDto(updateQuiz);
            repository.Update(quiz);
            return Task.CompletedTask;
        }

        public Task DeleteQuiz(Guid quizId)
        {
            var quiz = repository.Get(quizId);
            repository.Delete(quiz);
            return Task.CompletedTask;
        }
    }
}

File: Learnify.UI/src/services/QuizService.js
import { axios_config } from '../axios/axios_config';

const endpoint = '/quizzes'

export const getAllQuizzes = async () => {
    try {
        const response = await axios_config.get(endpoint);

        return response.status === 200 ? response.data : [];
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const getQuizById = async (quizId) => {
    try {
        const response = await axios_config.get(`${endpoint}/${quizId}`);

        return response.status === 200 ? response.data : null;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const getQuizWithQuestions = async (quizId) => {
    try {
        const response = await axios_config.get(`${endpoint}/${quizId}/questions`);

        return response.status === 200 ? response.data : null;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const getQuizzesByCourse = async (courseId) => {
    try {
        const response = await axios_config.get(`${endpoint}/course/${courseId}`);

        return response.status === 200 ? response.data : [];
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const insertQuiz = async (quizRequest) => {
    try {
        const response = await axios_config.post(endpoint, quizRequest);

        return response.status === 200;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const updateQuiz = async (updateQuiz) => {
    try {
        const response = await axios_config.put(endpoint, updateQuiz);

        return response.status === 200;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const deleteQuiz = async (quizId) => {
    try {
        const response = await axios_config.delete(`${endpoint}/${quizId}`);

        return response.status === 200;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};
